src/client/mapblock_mesh.cpp

```diff
@@ -585,6 +587,47 @@ void PartialMeshBuffer::draw(video::IVideoDriver *driver) const
 	MapBlockMesh
 */
 
+static void applyColorAndMerge(std::vector<PreMeshBuffer> &prebuffers)
+{
+	// TODO: we should change the meshgen so it already applies the tile color
+	// so that we don't need to this extra step.
+	// However currently the CAO code relies on the ability to erase the vertex
+	// colors (light data) before applying the tile colors.
+
+	for (auto &p : prebuffers) {
+		p.applyTileColor();
+		// erase color information for later comparisons
```

Comment:
Is this color information no longer relevant because the vertices are now colorized?
If so, then I would suggest to include a comment:

`The color information is now baked into 'p.vertices' and thus no longer relevant.`